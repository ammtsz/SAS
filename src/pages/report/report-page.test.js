import React from "react";
import Report from "./report-page.component";

import { render, screen, cleanup } from "@testing-library/react";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

import { MemoryRouter } from "react-router-dom";

describe("Page component <Report />", () => {
  const mockStore = configureMockStore();

  describe("Report created right after a quiz is finished", () => {
    const categoryId = 2;
    const categoryName = "Sports";

    const mockReports = {
      [categoryId]: {
        category_name: categoryName,
        results: {
          difficulty: {
            easy: { rights: 2, wrongs: 2 },
            medium: { rights: 2, wrongs: 2 },
            hard: { rights: 0, wrongs: 2 },
          },
          rights: 4,
          wrongs: 6,
        },
      },
    };
    const mockQuizCategory = {
      id: categoryId,
      name: categoryName,
      completed: 10,
    };
    const store = mockStore({
      categories: {
        reports: mockReports,
      },
      quiz: {
        category: mockQuizCategory,
      },
      report: {
        review: false,
        category: null,
      },
    });

    let component;

    beforeEach(() => {
      component = render(
        <Provider store={store}>
          <MemoryRouter>
            <Report />
          </MemoryRouter>
        </Provider>
      );
    });

    afterEach(() => {
      cleanup();
    });

    it("should render Report", () => {
      expect(component.asFragment()).toMatchSnapshot();
    });

    it("should contain 'congratulation' text", () => {
      expect(screen.getByText("Congratulation!")).toBeInTheDocument();
    });
  });

  describe("Report requested from categories page", () => {
    const categoryId = 1;
    const categoryName = "Board Games";

    const mockReports = {
      [categoryId]: {
        category_name: categoryName,
        results: {
          difficulty: {
            easy: { rights: 2, wrongs: 2 },
            medium: { rights: 2, wrongs: 2 },
            hard: { rights: 0, wrongs: 2 },
          },
          rights: 4,
          wrongs: 6,
        },
      },
    };
    const mockReportCategory = {
      id: categoryId,
      name: categoryName,
      completed: 10,
    };
    const store = mockStore({
      categories: {
        reports: mockReports,
      },
      quiz: {
        category: {},
      },
      report: {
        review: true,
        category: mockReportCategory,
      },
    });

    let component;

    beforeEach(() => {
      component = render(
        <Provider store={store}>
          <MemoryRouter>
            <Report />
          </MemoryRouter>
        </Provider>
      );
    });

    afterEach(() => {
      cleanup();
    });

    it("should render Report", () => {
      expect(component.asFragment()).toMatchSnapshot();
    });

    it("should contain category-name text", () => {
      expect(screen.getByText(categoryName)).toBeInTheDocument();
    });
  });
});
