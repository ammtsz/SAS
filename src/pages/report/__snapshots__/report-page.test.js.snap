// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page component <Report /> Report created right after a quiz is finished should render Report 1`] = `
<DocumentFragment>
  <section
    class="sc-dlfnbm dlSvaN"
    data-testid="report-page"
  >
    <div
      class="sc-bdfBwQ sc-hKgILt cXyUsK dBaUJn"
    >
      <div
        class="sc-eCssSg idQTZf"
      >
        <img
          class="sc-jSgupP iCuyco"
          src="mascot.svg"
        />
        <div
          class="sc-gKsewC jqveOd"
        >
          <h2
            class="sc-iBPRYJ eOmvXf"
          >
            Congratulation!
          </h2>
          <p
            class="sc-fubCfw hFwHhI"
          >
            You finished the quiz
          </p>
        </div>
      </div>
      <article
        class="sc-pFZIQ kpNfE"
      >
        <h4
          class="sc-jrAGrp yRYjm"
        >
          Check your performance
        </h4>
        <div
          class="sc-kEjbxe iykKIp"
          tabindex="0"
        >
          <div>
            <div
              class="sumary__number"
              data-testid="report-rights"
            >
              4
            </div>
            <div
              class="sumary__text"
            >
              Correct
            </div>
          </div>
          <div>
            <div
              class="sumary__number"
              data-testid="report-wrongs"
            >
              6
            </div>
            <div
              class="sumary__text"
            >
              Incorrect
            </div>
          </div>
        </div>
        <div
          class="sc-iqHYGH iQDmjP"
          tabindex="0"
        >
          <div
            class="level__details"
            data-testid="report-level-details"
          >
            <h6
              class="level__details--title"
            >
              Easy
            </h6>
            <p
              class="level__details--info"
              data-testid="report-easy-rights"
            >
              Correct: 2
            </p>
            <p
              class="level__details--info"
              data-testid="report-easy-wrongs"
            >
              Incorrect: 2
            </p>
          </div>
          <div
            class="level__details"
            data-testid="report-level-details"
          >
            <h6
              class="level__details--title"
            >
              Medium
            </h6>
            <p
              class="level__details--info"
              data-testid="report-medium-rights"
            >
              Correct: 2
            </p>
            <p
              class="level__details--info"
              data-testid="report-medium-wrongs"
            >
              Incorrect: 2
            </p>
          </div>
          <div
            class="level__details"
            data-testid="report-level-details"
          >
            <h6
              class="level__details--title"
            >
              Hard
            </h6>
            <p
              class="level__details--info"
              data-testid="report-hard-rights"
            >
              Correct: 0
            </p>
            <p
              class="level__details--info"
              data-testid="report-hard-wrongs"
            >
              Incorrect: 2
            </p>
          </div>
        </div>
        <button
          class="sc-gsTCUz sc-crrsfI dfZqOT cBQRTc"
        >
          Go to Homepage
        </button>
      </article>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Page component <Report /> Report requested from categories page should render Report 1`] = `
<DocumentFragment>
  <section
    class="sc-dlfnbm dlSvaN"
    data-testid="report-page"
  >
    <div
      class="sc-bdfBwQ sc-hKgILt cXyUsK dBaUJn"
    >
      <div
        class="sc-eCssSg idQTZf"
      >
        <img
          class="sc-jSgupP iCuyco"
          src="mascot.svg"
        />
        <div
          class="sc-gKsewC jqveOd"
        >
          <h2
            class="sc-iBPRYJ eOmvXf"
          >
            Board Games
          </h2>
          <p
            class="sc-fubCfw hFwHhI"
          />
        </div>
      </div>
      <article
        class="sc-pFZIQ kpNfE"
      >
        <h4
          class="sc-jrAGrp yRYjm"
        >
          Check your performance
        </h4>
        <div
          class="sc-kEjbxe iykKIp"
          tabindex="0"
        >
          <div>
            <div
              class="sumary__number"
              data-testid="report-rights"
            >
              4
            </div>
            <div
              class="sumary__text"
            >
              Correct
            </div>
          </div>
          <div>
            <div
              class="sumary__number"
              data-testid="report-wrongs"
            >
              6
            </div>
            <div
              class="sumary__text"
            >
              Incorrect
            </div>
          </div>
        </div>
        <div
          class="sc-iqHYGH iQDmjP"
          tabindex="0"
        >
          <div
            class="level__details"
            data-testid="report-level-details"
          >
            <h6
              class="level__details--title"
            >
              Easy
            </h6>
            <p
              class="level__details--info"
              data-testid="report-easy-rights"
            >
              Correct: 2
            </p>
            <p
              class="level__details--info"
              data-testid="report-easy-wrongs"
            >
              Incorrect: 2
            </p>
          </div>
          <div
            class="level__details"
            data-testid="report-level-details"
          >
            <h6
              class="level__details--title"
            >
              Medium
            </h6>
            <p
              class="level__details--info"
              data-testid="report-medium-rights"
            >
              Correct: 2
            </p>
            <p
              class="level__details--info"
              data-testid="report-medium-wrongs"
            >
              Incorrect: 2
            </p>
          </div>
          <div
            class="level__details"
            data-testid="report-level-details"
          >
            <h6
              class="level__details--title"
            >
              Hard
            </h6>
            <p
              class="level__details--info"
              data-testid="report-hard-rights"
            >
              Correct: 0
            </p>
            <p
              class="level__details--info"
              data-testid="report-hard-wrongs"
            >
              Incorrect: 2
            </p>
          </div>
        </div>
        <button
          class="sc-gsTCUz sc-crrsfI dfZqOT cBQRTc"
        >
          Go to Homepage
        </button>
      </article>
    </div>
  </section>
</DocumentFragment>
`;
